src/R/Components/Impl/ConnectionManager/Implementation/View/ConnectionManagerControl.xaml.cs

```diff
@@ -40,8 +40,17 @@ public partial class ConnectionManagerControl : UserControl {
         }
 
         private void ButtonEdit_Click(object sender, RoutedEventArgs e) {
-            Model?.Edit(GetConnection(e));
-            ScrollEditedIntoView();
+            if (Model != null) {
+                var currentConnection = Model.EditedConnection;
+                if (Model.IsEditingNew || currentConnection != null) {
+                    Model.CancelEdit();
+                }
```

Comment:
Let me check. The request was to close the panel (and cancel editing) when user clicks the button again.